evalml/automl/automl_search.py

```diff
@@ -674,8 +677,12 @@ def _compute_cv_scores(self, pipeline):
                 logger.debug(f"Skipping fold {i} because CV for stacked ensembles is not supported.")
                 break
             logger.debug(f"\t\tTraining and scoring on fold {i}")
-            X_train, X_valid = self.X_train.iloc[train], self.X_train.iloc[valid]
-            y_train, y_valid = self.y_train.iloc[train], self.y_train.iloc[valid]
+            if not self.sampler:
+                X_train, X_valid = self.X_train.iloc[train], self.X_train.iloc[valid]
+                y_train, y_valid = self.y_train.iloc[train], self.y_train.iloc[valid]
+            else:
+                X_train, y_train = train[0], train[1]
```

Comment:
The sampler splitter passes out the `[(X_train, y_train), (X_test, y_test)]` data as tuples, not indices. We can't pass out indices with `SMOTE`, which generates synthetic data